Lib/asyncio/coroutines.py

```diff
@@ -13,14 +14,10 @@ def _is_debug_mode():
                                   bool(os.environ.get('PYTHONASYNCIODEBUG')))
 
 
-# A marker for iscoroutinefunction.
-_is_coroutine = object()
-
-
 def iscoroutinefunction(func):
-    """Return True if func is a decorated coroutine function."""
-    return (inspect.iscoroutinefunction(func) or
-            getattr(func, '_is_coroutine', None) is _is_coroutine)
+    """Alias for inspect.iscoroutinefunction."""
+    warnings._deprecated("asyncio.iscoroutinefunction", remove=(3, 14))
```

Comment:
I'll need to do a doc change for this too

Reply:
cc @hugovk for removals

Reply:
Please could you add it to a 3.14 section like https://docs.python.org/3.12/whatsnew/3.12.html#pending-removal-in-python-3-13 ?

And add a `.. deprecated-removed:: 3.12 3.14` to the function in its module rst page.